frontend/src/utils/replace_content_with_refs.ts

```
@@ -105,21 +110,37 @@ function replace_nested_nodes(
-  for (let i = 0; i < curr_node.schema.length; i++) {
-    give_new_ids_helper(curr_node.schema, curr_node.schema[i], i);
-  }
-}
+// This is still a wildly innefficient way of solving this problem. We would
+// like to not have to replace the id's. We are cloning objects in the structuredClone
+// call in replace_nested_nodes. For viewing subtrees we need to ensure that
+// all id's are unique. Iterative > recursive for this, but not existing is
+// better.
+function give_new_ids(curr_node: DBDetails): void {
+  const que: ColumnSchema[] = [...curr_node.schema];
+  let current_q_idx: number = 0;
```

Biome (1.9.4)

[error] This type annotation is trivially inferred from its initialization.

Safe fix: Remove the type annotation.


(lint/style/noInferrableTypes)